Sources/_StringProcessing/Algorithms/Consumers/PredicateConsumer.swift

```diff
@@ -0,0 +1,42 @@
+public struct PredicateConsumer<Consumed: Collection> {
```

Comment:
Why is this public? It serves as an interesting example of a trivial conformer and a nice way to test that composition and layering work, but IIUC I wouldn't expect people to explicitly make one or for it to appear in API.

Reply:
It's currently used in the return type of a couple `split` overloads, so I had to make this type public or wrap those return types into new public types and make this one private. I went with the easy route initially but I can certainly wrap it if we think that's worth it. More powerful opaque return types will solve this too, so that is mainly why I didn't bother.

Reply:
Makes sense. Opaque return types will help, and we'll see what ends up in the `Swift` default import vs outside.